apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.RestoreObject.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Object = Google.Apis.Storage.v1.Data.Object;
+
+namespace Google.Cloud.Storage.V1;
+public abstract partial class StorageClient
+{
+    /// <summary>
+    /// Restores a soft-deleted object.
+    /// </summary>
+    /// <param name="bucket">The name of the bucket containing the object. Must not be null.</param>
+    /// <param name="objectName">The name of the object to restore. Must not be null.</param>
+    /// <param name="generation">The specific revision of the object to restore.</param>
```

Comment:
Does Storage use the term "revision"? Maybe we should use "generation" here instead.

Reply:
The underlying docs read: "Selects a specific revision of this object." and the underlying parameter is named "generation" :(

They may be used interchangeably by Storage? I'm happy yo change to generation if you think it better?

Reply:
Let's stick with the underlying docs for consistency with them (even if they're inconsistent themselves).